core/tests/puppeteer-acceptance-tests/user-utilities/curriculum-admin-utils.ts

```diff
@@ -0,0 +1,703 @@
+
+const assignSkillButton = 'i.e2e-test-skill-item-edit-btn';
+const confirmSkillAssignationButton =
+  'button.e2e-test-skill-assign-subtopic-confirm';
+
+const addDiagnosticTestSkillButton =
+  'button.e2e-test-add-diagnostic-test-skill';
+const diagnosticTestSkillSelector =
+  'select.e2e-test-diagnostic-test-skill-selector';
+
+const saveChangesMessageInput = 'textarea.e2e-test-commit-message-input';
+
+const addStoryButton = 'button.e2e-test-create-story-button';
+const storyTitleField = 'input.e2e-test-new-story-title-field';
+const storyDescriptionField = 'textarea.e2e-test-new-story-description-field';
+const storyUrlFragmentField = 'input.e2e-test-new-story-url-fragment-field';
+const createStoryButton = 'button.e2e-test-confirm-story-creation-button';
+const saveStoryButton = 'button.e2e-test-save-story-button';
+const publishStoryButton = 'button.e2e-test-publish-story-button';
+const storyMetaTagInput = '.e2e-test-story-meta-tag-content-field';
+const unpublishStoryButton = 'button.e2e-test-unpublish-story-button';
+
+const addChapterButton = 'button.e2e-test-add-chapter-button';
+const chapterTitleField = 'input.e2e-test-new-chapter-title-field';
+const chapterExplorationIdField = 'input.e2e-test-chapter-exploration-input';
+const createChapterButton = 'button.e2e-test-confirm-chapter-creation-button';
+
+const explorationMobileSaveMenu = '.e2e-test-mobile-options';
+const explorationMobileSaveButton = '.e2e-test-save-changes-for-small-screens';
+
+const mobileToastMessage = '.e2e-test-toast-message';
+const mobileOptionsSelector = '.e2e-test-mobile-options-base';
+const mobileSaveChangesDropdown = '.e2e-test-mobile-changes-dropdown';
+const mobilePublishButton = '.e2e-test-mobile-publish-button';
+
+const mobileTopicSelector = '.e2e-test-mobile-topic-name';
+const mobileSkillsTab = '.e2e-test-mobile-skills-tab';
+const mobileSaveTopicButton =
+  'div.navbar-mobile-options .e2e-test-mobile-save-topic-button';
+const mobileStoryDropdown = '.e2e-test-story-dropdown';
+const mobileSaveStoryChangesDropdown =
+  'div.navbar-mobile-options .e2e-test-mobile-changes-dropdown';
+const mobileSaveStoryChangesButton =
+  'div.navbar-mobile-options .e2e-test-mobile-save-changes';
+const mobilePublishStoryButton =
+  'div.navbar-mobile-options .e2e-test-mobile-publish-button';
+const mobileAddChapterDropdown = '.e2e-test-mobile-add-chapter';
+
+export class CurriculumAdmin extends BaseUser {
+  /**
+   * Function for navigating to the topic and skills dashboard page.
+   */
+  async navigateToTopicAndSkillsDashboardPage(): Promise<void> {
+    await this.page.bringToFront();
+    await this.goto(topicAndSkillsDashboardUrl);
+  }
+
+  /**
+   * Function for creating a skill in the topics and skills dashboard.
+   * We need to generate at least 3 questions for the skill as a
+   * pre-requisite for publishing the topic later.
+   */
+  async createSkill(description: string, questionCount: number): Promise<void> {
```

Comment:
A further thought about this -- I was reviewing another acceptance test PR and wondering why I didn't feel the same way about the functions in here: https://github.com/oppia/oppia/pull/19780/files#diff-885139b0d44e4d97ca4c17e8b918ed7a53dcdcc08814a1b3728a625a2c8542c4R92

I think it's because, in the other PR, it is easy to tell what the functions are doing **on the page**. So, if a test attempts e.g. switchToEditorTab() in a context that doesn't make sense, then it's easy to identify that. With createSkill() it is much harder (I can't mentally map that to an action in the UI). 

So just noting that a third alternative to the above is to name the function clearly so that it's possible to easily visualize what it does.